tutorial2.md

```diff
@@ -0,0 +1,137 @@
+  /kubeadm                   Uploads the kubeadm ClusterConfiguration to a ConfigMap
+  /kubelet                   Uploads the kubelet component config to a ConfigMap
+mark-control-plane         Mark a node as a control-plane
+bootstrap-token            Generates bootstrap tokens used to join a node to a cluster
+addon                      Installs required addons for passing Conformance tests
+  /coredns                   Installs the CoreDNS addon to a Kubernetes cluster
+  /kube-proxy                Installs the kube-proxy addon to a Kubernetes cluster
+
+
+Usage:
+  kubeadm init [flags]
+  kubeadm init [command]
+
+Available Commands:
+  phase       use this command to invoke single phase of the init workflow
+
+
+<snip>
+```
+
+For now, we're just going to run the preflight checks.
+
+```shell
+kubeadm init phase preflight
+```
+
+Depending on how you've set up your VM, you might see one of several errors.
+Here's a breakdown.
+
+### IsPrivilegedUser
+
+Error: `[ERROR IsPrivilegedUser]: user is not running as root`
+
+Fix: Run `kubeadm` with `sudo`
+
+### Swap
+
+Error: `[ERROR Swap]: running with swap on is not supported. Please disable swap`
+
+Fix: `sudo swapoff -a`. You can also remove the swap mount from `/etc/fstab` for a more permanent solution
+
+### SystemVerification
+
+Error: `[WARNING SystemVerification]: this Docker version is not on the list of validated versions: 18.09.2. Latest validated version: 18.06`
+
+Fix: You can downgrade Docker to a supported version.
+List the available versions:
+
+``` console
+$ sudo apt-cache madison docker.io
+ docker.io | 18.09.2-0ubuntu1~18.04.1 | http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages
+ docker.io | 18.06.1-0ubuntu1.2~18.04.1 | http://archive.ubuntu.com/ubuntu bionic-security/universe amd64 Packages
+ docker.io | 17.12.1-0ubuntu1 | http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
+ ```
+
+That second one is the one we want. You specify a version for `apt` using `=`:
+
+```console
+sudo apt install docker.io=18.06.1-0ubuntu1.2~18.04.1
+```
+
+If you have containers running already, a prompt will ask you if you want to restart the docker service (and therefore them). Do this, as the kubelet will handle restarting your containers.
+
+### NumCPU
+
+Error: `[ERROR NumCPU]: the number of available CPUs 1 is less than the required 2`
+
+Fix: By default, Virtualbox gives VMs one CPU.
+To mollify kubelet, we can increase this to two.
+
+1. Shut down your VM. It needs to actually shut down, not just be paused.
+
+   ```console
+   sudo systemctl poweroff
+   ```
+
+2. go into the Virtualbox Manager window (i.e. the one that lists all your VMs.)
+3. Select your k8s virtual machine, and click Settings.
+4. Click the System option on the left.
+5. Under the Processor tab, increase the slider to two.
+
+That's it! Reboot your machine, and it should pass this preflight check now.
+
```

Comment:
two more errors not mentioned here:

elizabeth@lily:~$ sudo kubeadm init phase preflight
[preflight] Running pre-flight checks
        [WARNING Service-Docker]: docker service is not enabled, please run 'systemctl enable docker.service'
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR Port-10250]: Port 10250 is in use

elizabeth@lily:~$ sudo netstat -anp |grep -i 10250
tcp6       0      0 :::10250                :::*                    LISTEN      10569/kubelet